src/com/google/enterprise/adaptor/testing/RecordingResponse.java

```diff
@@ -277,12 +280,103 @@ public boolean isSecure() {
   }
 
   /**
-   * Gets an unmodifiable list of the accumulated anchors.
+   * Gets an unmodifiable multimap of the accumulated anchors.
    *
-   * @return a unmodifiable list of the accumulated anchors
+   * @return a unmodifiable multimap of the accumulated anchors
+   */
+  public AnchorMap getAnchors() {
+    return new AnchorMap(anchors);
+  }
+
+  /**
+   * An immutable multimap that supports duplicate, ordered key-value pairs.
+   * The anchor texts are the keys, and the anchor URIs are the values.
+   * <p>
+   * This class is consistent with Guava's {@code LinkedListMultimap},
+   * although it does not extend that class, or implement the
+   * {@code ListMultimap} interface or all of its methods.
    */
-  public List<Map.Entry<String, URI>> getAnchors() {
-    return unmodifiableList(anchors);
+  public static class AnchorMap {
+    private final List<Map.Entry<String, URI>> anchors;
+
+    private AnchorMap(List<Map.Entry<String, URI>> anchors) {
+      this.anchors = unmodifiableList(anchors);
```

Comment:
do we need a copy?  if someone calls addAnchor after getAnchors their List, that they got before, will change.  Is that OK?

Reply:
I think it is, or at least, it's certainly consistent. All of the methods in RecordingResponse that return collections now return unmodifiable views on the underlying data, which can still be changed via the methods of the Response interface. I think we should just make that more clear in the javadoc.

Reply:
I don't think we're being fast-and-loose, or writing lesser code because it's aimed at tests. Providing unmodifiable views is different from defensive copies because we're the producer of the data, not the consumer. Making copies would impose an unnecessary decision on the caller. If the caller wants to make sure that the values don't change, then a defensive copy would be reasonable on their side. Our task is simply to make sure we only reflect changes made through the Response interface, and make sure callers don't have direct access to editing the recorded data.

As you suggest, though, with simple, single-threaded tests, the semantic difference is not noticeable.

These classes were an amalgam of half a dozen copies spread through the library (which had more than one) and various connectors. The copies were inconsistent about whether they provided the actual collection references or an unmodifiable view, and I chose the latter when they conflicted. A previous change (#86) made RecordingResponse return unmodifiable views like RecordingDocIdPusher already did, including the anchors.